src/world.cpp

```diff
@@ -2521,6 +2521,11 @@ void World_GenRoomCollision()
 {
     room_p r = global_world.rooms;
 
+    if(r == NULL)
+    {
+        return;
```

Comment:
if(r == NULL) then return - is normal condition;
assert(rooms == NULL || rooms_count > 0); - there are cases engine try to load level that not exists, so runtime checks is needed here + some original levels I load contains only textures (TR5)